src/Hosting/Hosting/src/Startup/ConventionBasedStartup.cs

```diff
@@ -18,39 +18,15 @@ public ConventionBasedStartup(StartupMethods methods)
         {
             _methods = methods;
         }
-        
+
         public void Configure(IApplicationBuilder app)
         {
-            try
-            {
-                _methods.ConfigureDelegate(app);
-            }
-            catch (Exception ex)
-            {
-                if (ex is TargetInvocationException)
-                {
-                    ExceptionDispatchInfo.Capture(ex.InnerException).Throw();
-                }
-
-                throw;
-            }
+            _methods.ConfigureDelegate(app);
         }
 
         public IServiceProvider ConfigureServices(IServiceCollection services)
         {
-            try
-            {
-                return _methods.ConfigureServicesDelegate(services);
-            }
-            catch (Exception ex)
-            {
-                if (ex is TargetInvocationException)
```

Comment:
Isn't the TargetInvocationException still possible for ConfigureServices when you fall back to MethodInfo.Invoke?